package/src/handler.ts

```diff
@@ -669,4 +675,48 @@ export class Handler {
             ),
         })
     }
+
+    /**
+     * Highlights lines that contain symbol definitions in red.
+     */
+    async debugAnnotate(doc: TextDocument): Promise<void> {
+        if (this.debugAnnotatedURIs.includes(doc.uri)) {
+            return
+        }
+        this.debugAnnotatedURIs.push(doc.uri)
+        setTimeout(async () => {
+            const editor = this.sourcegraph.app.activeWindow
+                ? this.sourcegraph.app.activeWindow.visibleViewComponents[0]
+                : undefined
+            if (!editor) {
+                console.log('NO EDITOR')
+            } else {
+                const { repo, rev, path } = parseUri(doc.uri)
+
+                const r = await this.api.search(
+                    `repo:${repo}@${rev} file:${path} type:symbol ^` // ^ matches everything (can't leave out a query)
+                )
+                editor.setDecorations(
+                    this.sourcegraph.app.createDecorationType(),
```

Comment:
This decoration type should be created at the top-level and reused, not generated every time `setDecorations()` is called.